ocaml/xapi/xapi_pbd.ml

```diff
@@ -247,10 +245,18 @@ let unplug_all_pbds ~__context =
          TaskHelper.exn_if_cancelling ~__context;
          debug "Unplugging PBD %s" uuid;
          unplug ~__context ~self:pbd);
-  debug "Finished unplug_all_pbds";
+  debug "Finished unplug_all_pbds"
+
+let disable_clustering ~__context =
   let host = Helpers.get_localhost ~__context in
   match Xapi_clustering.find_cluster_host ~__context ~host with
   | None -> info "No cluster host found"
   | Some self ->
      info "Disabling cluster host";
      Xapi_cluster_host.disable ~__context ~self
+
+(* Called on shutdown: it unplugs all the PBDs and disables the cluster host.
+   If anything fails it throws an exception *)
+let unplug_all_pbds ~__context =
+  unplug_all_pbds_internal ~__context;
+  disable_clustering ~__context
```

Comment:
Please change all the caller of unplug_all_pbds to call these two functions instead. The reason behind splitting the function was to make it clearer what it does: a function called unplug_all_pbds also disabling clustering is somewhat unexpected.